build-tools/packages/build-tools/src/test/biomeConfig.test.ts

```diff
@@ -0,0 +1,214 @@
+			});
+
+			it("linter.ignore has correct value", async () => {
+				const testFile = path.resolve(testDataPath, "biome/pkg-a/biome.jsonc");
+				const actual = await loadBiomeConfig(testFile);
+
+				assert(actual.linter?.ignore?.includes("ignore-linter/**"));
+				assert(actual.linter?.ignore?.length === 1);
+			});
+		});
+	});
+
+	describe("getSettingValuesFromBiomeConfig", async () => {
+		describe("extends from a single config", async () => {
+			let testConfig: BiomeConfig;
+			before(async () => {
+				const testFile = path.resolve(testDataPath, "biome/pkg-a/biome.jsonc");
+				testConfig = await loadBiomeConfig(testFile);
+			});
+
+			it("formatter ignore settings are merged with root", async () => {
+				const ignores = await getSettingValuesFromBiomeConfig(
+					testConfig,
+					"formatter",
+					"ignore",
+				);
+				assert(ignores.has("pkg-a-ignore/**"));
+				assert(ignores.has("ignore-formatter/**"));
+				assert(ignores.size === 2);
+			});
+
+			it("linter ignore settings are merged with root", async () => {
+				const ignores = await getSettingValuesFromBiomeConfig(testConfig, "linter", "ignore");
+				assert(ignores.has("pkg-a-ignore/**"));
+				assert(ignores.has("ignore-linter/**"));
+				assert(ignores.size === 2);
+			});
+
+			it("formatter include settings are merged with root", async () => {
+				const includes = await getSettingValuesFromBiomeConfig(
+					testConfig,
+					"formatter",
+					"include",
+				);
+				assert(includes.has("pkg-a-include/**"));
+				assert(includes.has("include-formatter/**"));
+				assert(includes.size === 2);
+			});
+
+			it("linter include settings are merged with root", async () => {
+				const includes = await getSettingValuesFromBiomeConfig(
+					testConfig,
+					"linter",
+					"include",
+				);
+				assert(includes.has("pkg-a-include/**"));
+				assert(includes.has("include-linter/**"));
+				assert(includes.size === 2);
+			});
+		});
+	});
+
+	describe("getBiomeFormattedFiles", async () => {
+		describe("extends from a single config", async () => {
+			let testPath: string;
+			let gitRepo: GitRepo;
+			before(async () => {
+				const repoRoot = await getResolvedFluidRoot(true);
+				gitRepo = new GitRepo(repoRoot);
+				testPath = path.resolve(testDataPath, "biome/pkg-a/");
+			});
+
+			it("returns correct file set", async () => {
+				const expected = [
+					path.resolve(testDataPath, "biome/pkg-a/pkg-a-include/sourceFile.ts"),
+					path.resolve(
+						testDataPath,
+						"biome/pkg-a/pkg-a-include/include-formatter/formatter.ts",
+					),
+					path.resolve(testDataPath, "biome/pkg-a/pkg-a-include/include-linter/linter.ts"),
+					path.resolve(testDataPath, "biome/pkg-a/include-formatter/formatter.ts"),
+				];
+				const formattedFiles = await getBiomeFormattedFiles(testPath, gitRepo);
+				for (const actual of formattedFiles) {
+					assert(expected.includes(actual));
+				}
+				assert(formattedFiles.length === 4);
```

Comment:
messages are good for otherwise very plain checks. I think what you mean to check here is that no extra files were identified.

Reply:
Yes, these length checks are a kludge. I really want something like `expect(actual).toIncludeOnly(expected)`, but I don't want to add deps on chai/chai-arrays to build-tools. Reason is that I want to move away from mocha/chai altogether (see #21920). I'll add a message to these length checks to make it clearer.